File: src/main/java/com/dietagent/dto/response/NutritionSummaryResponse.java
package com.dietagent.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class NutritionSummaryResponse {

    private Integer totalCalories;
    private Double totalProtein;
    private Double totalFat;
    private Double totalCarbohydrate;

    private Integer calorieGoal;
    private Integer proteinGoal;
    private Integer fatGoal;
    private Integer carbGoal;

    private Double calorieProgress;
    private Double proteinProgress;
    private Double fatProgress;
    private Double carbProgress;
}

File: android/app/src/main/java/com/dietagent/android/data/remote/dto/NutritionSummaryResponse.kt
package com.dietagent.android.data.remote.dto

import kotlinx.serialization.Serializable

@Serializable
data class NutritionSummaryResponse(
    val totalCalories: Int? = 0,
    val totalProtein: Double? = 0.0,
    val totalFat: Double? = 0.0,
    val totalCarbohydrate: Double? = 0.0,
    val calorieGoal: Int? = 0,
    val proteinGoal: Int? = 0,
    val fatGoal: Int? = 0,
    val carbGoal: Int? = 0,
    val calorieProgress: Double? = 0.0,
    val proteinProgress: Double? = 0.0,
    val fatProgress: Double? = 0.0,
    val carbProgress: Double? = 0.0,
)
